### PR TITLE
add a 'toDbVersion' helper, other db testing helpers

### PR DESCRIPTION
**Pull request by djmitche**
pulled out of #2932

edit: whoops, I failed at `git add -p`.  This has a few more things in it.

### CODE REVIEW

**Comment by djmitche** [comment]
* add a 'toDbVersion' helper function to upgrade/downgrade as necessary
* testEntityTable now supports a second customTests argument to add
      additional non-automated tests that still get to use the sample data
* suites are more particular about DB versions, for more accurate tests
      when using `mocha --grep`

File: db/test/helper.js [hunk 1]
```diff
@@ -271,9 +288,10 @@ exports.testEntityTable = ({
   // it is treated as a collection of modifier functions to run in parallel to check
   // support for concurrent modifications.
   modifications,
-}) => {
+  // customTests can define additional test cases in the usual Mocha style.  The
+  // parameter is true if the tests are run for THIS_VERSION, otherwise for PREV_VERSION.
+}, customTests = (isThisVersion) => {}) => {
```
Comment (helfi92) [inline db/test/helper.js:293]: `isThisVersion` looks like it's currently unused.

Comment (djmitche) [reply db/test/helper.js:293]: It's used in test cases where customTests is given (#3115).  It's here just to illustrate.

**Comment by djmitche** [comment]
> It would probably be a good idea to add a tiny section about this in the migration document.

Good point.  The uuid functions, too.

**Pull request merged**